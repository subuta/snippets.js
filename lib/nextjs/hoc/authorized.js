import { build, snippets as s } from 'bld.js'
import _ from 'lodash'

export default () => {
  const imports = s.import([
    ['react', 'React'],
    ['lodash', '_'],
    ['src/views/utils/auth0', 'auth0']
  ])

  return build`
    ${imports}
    
    export default function (Component) {
      let render = (props) => {
        return (
          <Component {...props} />
        )
      }
    
      render.getInitialProps = async (ctx) => {
        const fn = Component.getInitialProps || _.noop
    
        const isAuthenticated = auth0.isAuthenticated(ctx)
    
        // decorate ctx.
        if (!!ctx.res) {
          const locals = _.get(ctx, 'res.locals', {})
          _.set(ctx, 'res.locals', {
            ...locals,
            isAuthenticated
          })
        }
    
        const props = await fn(ctx)
    
        // return final props.
        return {
          ...props,
          isAuthenticated
        }
      }
    
      return render
    }

  `
}
