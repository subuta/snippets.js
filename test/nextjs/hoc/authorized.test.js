/* global expect, describe, it */

import { build, format, snippets as s } from 'bld.js'

import authorized from 'lib/nextjs/hoc/authorized'

describe('nextjs/hoc/authorized', () => {
  it('should create Authorized hoc for next.js', () => {
    expect(format(authorized())).toMatchSnapshot()
  })
})
