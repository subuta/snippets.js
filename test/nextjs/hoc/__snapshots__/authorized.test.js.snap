// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nextjs/hoc/authorized should create Authorized hoc for next.js 1`] = `
"import React from 'react'
import _ from 'lodash'
import auth0 from 'src/views/utils/auth0'

export default function(Component) {
  let render = (props) => {
    return <Component {...props} />
  }

  render.getInitialProps = async (ctx) => {
    const fn = Component.getInitialProps || _.noop

    const isAuthenticated = auth0.isAuthenticated(ctx)

    // decorate ctx.
    if (!!ctx.res) {
      const locals = _.get(ctx, 'res.locals', {})
      _.set(ctx, 'res.locals', {
        ...locals,
        isAuthenticated
      })
    }

    const props = await fn(ctx)

    // return final props.
    return {
      ...props,
      isAuthenticated
    }
  }

  return render
}
"
`;
